### PR TITLE
fix: `find_latest_share_counter` finds the last share count

### LINKED ISSUE #196: `find_share` should return the last count
# Summary

## Problem

`find_share` returns the first count.

### Impact

Can't figure out the counter needed to make a new share.

## Solution

There should be a function that returns the counter needed to make a new share.

### PR DESCRIPTION
**Pull request by matheus23**
## Summary

Fixes #196 

The idea is to provide a function that finds the maximum share count for available shares. This allows callers to check all share counts they might have missed in between.

## Test plan (required)

I added another test that checks after a bunch of shares that the maximum share counter is returned.

### CODE REVIEW

**Comment by matheus23** [comment]
Okay, fixed the wasm. CI is green now :)

**Review by appcypher — APPROVED** [review]
This looks great to me. I only have minor suggestions that don't really affect the code.
Thanks for fixing this!

File: wnfs/src/private/share.rs [hunk 1]
```diff
@@ -317,31 +317,38 @@ pub mod recipient {
...
         sharer_forest: &PrivateForest,
         store: &impl BlockStore,
     ) -> Result<Option<u64>> {
         for i in 0..limit {
-            let share_label = sharer::create_share_label(
-                share_count + i,
-                sharer_root_did,
-                recipient_exchange_key,
-            );
+            let share_count = share_count_start + i;
```
Comment (appcypher) [inline wnfs/src/private/share.rs:330]: Nit: Not that it matters much

```rs
for share_count in share_count_start..(share_count_start + limit) {
    // ...
```

File: wnfs/src/private/share.rs [hunk 2]
```diff
@@ -317,31 +317,38 @@ pub mod recipient {
...
             {
-                return Ok(Some(share_count + i));
+                if i == 0 {
+                    // There don't seem to be any shares yet
+                    return Ok(None);
+                } else {
+                    // We've hit the first unpopulated label,
+                    // so the last one was the last valid share counter.
+                    return Ok(Some(share_count - 1));
+                }
```
Comment (appcypher) [inline wnfs/src/private/share.rs:346]: Nit: Cosmetic suggestion :)

```rs
return Ok(...)
```

File: wnfs/src/private/share.rs (hunk 1 above)
Comment (matheus23) [reply wnfs/src/private/share.rs:330]: I like it :) Will update :+1:

File: wnfs/src/private/share.rs (hunk 2 above)
Comment (matheus23) [reply wnfs/src/private/share.rs:346]: Hm, I don't understand?

Comment (matheus23) [reply wnfs/src/private/share.rs:346]: Are you referring to the last line? clippy actually complains if I put `return Ok(Some(...));` into the last line instead of `Ok(Some(...))`.

Comment (appcypher) [reply wnfs/src/private/share.rs:346]: Sorry. I mean 

```rs
return Ok(if i == 0 {
    // There don't seem to be any shares yet
    None
} else {
    // We've hit the first unpopulated label,
    // so the last one was the last valid share counter.
    Some(share_count - 1)
})
```

You can ignore if it doesn't appeal to your senses.

Comment (matheus23) [reply wnfs/src/private/share.rs:346]: Oh. I actually had it like this before, but ended up changing it.
Especially with the above change it's going to be a bigger check than `i == 0`, so perhaps more readable if it's not an expression :D
I'll keep it this way.

Comment (appcypher) [reply wnfs/src/private/share.rs]: That makes sense
